// react-peel/dist/react-peel.es.js
import Jt, { forwardRef as Ct, useRef as Ut, useEffect as Q } from "react";
var jt = { exports: {} }, yt = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var qt;
function de() {
  if (qt)
    return yt;
  qt = 1;
  var b = Jt, _ = Symbol.for("react.element"), D = Symbol.for("react.fragment"), s = Object.prototype.hasOwnProperty, G = b.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, V = { key: !0, ref: !0, __self: !0, __source: !0 };
  function y(I, v, E) {
    var L, k = {}, W = null, q = null;
    E !== void 0 && (W = "" + E), v.key !== void 0 && (W = "" + v.key), v.ref !== void 0 && (q = v.ref);
    for (L in v)
      s.call(v, L) && !V.hasOwnProperty(L) && (k[L] = v[L]);
    if (I && I.defaultProps)
      for (L in v = I.defaultProps, v)
        k[L] === void 0 && (k[L] = v[L]);
    return { $$typeof: _, type: I, key: W, ref: q, props: k, _owner: G.current };
  }
  return yt.Fragment = D, yt.jsx = y, yt.jsxs = y, yt;
}
var mt = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Xt;
function pe() {
  return Xt || (Xt = 1, process.env.NODE_ENV !== "production" && function() {
    var b = Jt, _ = Symbol.for("react.element"), D = Symbol.for("react.portal"), s = Symbol.for("react.fragment"), G = Symbol.for("react.strict_mode"), V = Symbol.for("react.profiler"), y = Symbol.for("react.provider"), I = Symbol.for("react.context"), v = Symbol.for("react.forward_ref"), E = Symbol.for("react.suspense"), L = Symbol.for("react.suspense_list"), k = Symbol.for("react.memo"), W = Symbol.for("react.lazy"), q = Symbol.for("react.offscreen"), w = Symbol.iterator, M = "@@iterator";
    function tt(r) {
      if (r === null || typeof r != "object")
        return null;
      var o = w && r[w] || r[M];
      return typeof o == "function" ? o : null;
    }
    var $ = b.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function A(r) {
      {
        for (var o = arguments.length, a = new Array(o > 1 ? o - 1 : 0), f = 1; f < o; f++)
          a[f - 1] = arguments[f];
        xt("error", r, a);
      }
    }
    function xt(r, o, a) {
      {
        var f = $.ReactDebugCurrentFrame, m = f.getStackAddendum();
        m !== "" && (o += "%s", a = a.concat([m]));
        var P = a.map(function(g) {
          return String(g);
        });
        P.unshift("Warning: " + o), Function.prototype.apply.call(console[r], console, P);
      }
    }
    var X = !1, Tt = !1, st = !1, _t = !1, Lt = !1, ut;
    ut = Symbol.for("react.module.reference");
    function ct(r) {
      return !!(typeof r == "string" || typeof r == "function" || r === s || r === V || Lt || r === G || r === E || r === L || _t || r === q || X || Tt || st || typeof r == "object" && r !== null && (r.$$typeof === W || r.$$typeof === k || r.$$typeof === y || r.$$typeof === I || r.$$typeof === v || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      r.$$typeof === ut || r.getModuleId !== void 0));
    }
    function lt(r, o, a) {
      var f = r.displayName;
      if (f)
        return f;
      var m = o.displayName || o.name || "";
      return m !== "" ? a + "(" + m + ")" : a;
    }
    function nt(r) {
      return r.displayName || "Context";
    }
    function N(r) {
      if (r == null)
        return null;
      if (typeof r.tag == "number" && A("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof r == "function")
        return r.displayName || r.name || null;
      if (typeof r == "string")
        return r;
      switch (r) {
        case s:
          return "Fragment";
        case D:
          return "Portal";
        case V:
          return "Profiler";
        case G:
          return "StrictMode";
        case E:
          return "Suspense";
        case L:
          return "SuspenseList";
      }
      if (typeof r == "object")
        switch (r.$$typeof) {
          case I:
            var o = r;
            return nt(o) + ".Consumer";
          case y:
            var a = r;
            return nt(a._context) + ".Provider";
          case v:
            return lt(r, r.render, "ForwardRef");
          case k:
            var f = r.displayName || null;
            return f !== null ? f : N(r.type) || "Memo";
          case W: {
            var m = r, P = m._payload, g = m._init;
            try {
              return N(g(P));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var U = Object.assign, j = 0, et, ft, ht, bt, rt, Pt, wt;
    function J() {
    }
    J.__reactDisabledLog = !0;
    function dt() {
      {
        if (j === 0) {
          et = console.log, ft = console.info, ht = console.warn, bt = console.error, rt = console.group, Pt = console.groupCollapsed, wt = console.groupEnd;
          var r = {
            configurable: !0,
            enumerable: !0,
            value: J,
            writable: !0
          };
          Object.defineProperties(console, {
            info: r,
            log: r,
            warn: r,
            error: r,
            group: r,
            groupCollapsed: r,
            groupEnd: r
          });
        }
        j++;
      }
    }
    function u() {
      {
        if (j--, j === 0) {
          var r = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: U({}, r, {
              value: et
            }),
            info: U({}, r, {
              value: ft
            }),
            warn: U({}, r, {
              value: ht
            }),
            error: U({}, r, {
              value: bt
            }),
            group: U({}, r, {
              value: rt
            }),
            groupCollapsed: U({}, r, {
              value: Pt
            }),
            groupEnd: U({}, r, {
              value: wt
            })
          });
        }
        j < 0 && A("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var pt = $.ReactCurrentDispatcher, F;
    function it(r, o, a) {
      {
        if (F === void 0)
          try {
            throw Error();
          } catch (m) {
            var f = m.stack.trim().match(/\n( *(at )?)/);
            F = f && f[1] || "";
          }
        return `
` + F + r;
      }
    }
    var K = !1, Y;
    {
      var Et = typeof WeakMap == "function" ? WeakMap : Map;
      Y = new Et();
    }
    function O(r, o) {
      if (!r || K)
        return "";
      {
        var a = Y.get(r);
        if (a !== void 0)
          return a;
      }
      var f;
      K = !0;
      var m = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var P;
      P = pt.current, pt.current = null, dt();
      try {
        if (o) {
          var g = function() {
            throw Error();
          };
          if (Object.defineProperty(g.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(g, []);
            } catch (Z) {
              f = Z;
            }
            Reflect.construct(r, [], g);
          } else {
            try {
              g.call();
            } catch (Z) {
              f = Z;
            }
            r.call(g.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (Z) {
            f = Z;
          }
          r();
        }
      } catch (Z) {
        if (Z && f && typeof Z.stack == "string") {
          for (var p = Z.stack.split(`
`), B = f.stack.split(`
`), S = p.length - 1, R = B.length - 1; S >= 1 && R >= 0 && p[S] !== B[R]; )
            R--;
          for (; S >= 1 && R >= 0; S--, R--)
            if (p[S] !== B[R]) {
              if (S !== 1 || R !== 1)
                do
                  if (S--, R--, R < 0 || p[S] !== B[R]) {
                    var z = `
` + p[S].replace(" at new ", " at ");
                    return r.displayName && z.includes("<anonymous>") && (z = z.replace("<anonymous>", r.displayName)), typeof r == "function" && Y.set(r, z), z;
                  }
                while (S >= 1 && R >= 0);
              break;
            }
        }
      } finally {
        K = !1, pt.current = P, u(), Error.prepareStackTrace = m;
      }
      var at = r ? r.displayName || r.name : "", Ht = at ? it(at) : "";
      return typeof r == "function" && Y.set(r, Ht), Ht;
    }
    function d(r, o, a) {
      return O(r, !1);
    }
    function t(r) {
      var o = r.prototype;
      return !!(o && o.isReactComponent);
    }
    function e(r, o, a) {
      if (r == null)
        return "";
      if (typeof r == "function")
        return O(r, t(r));
      if (typeof r == "string")
        return it(r);
      switch (r) {
        case E:
          return it("Suspense");
        case L:
          return it("SuspenseList");
      }
      if (typeof r == "object")
        switch (r.$$typeof) {
          case v:
            return d(r.render);
          case k:
            return e(r.type, o, a);
          case W: {
            var f = r, m = f._payload, P = f._init;
            try {
              return e(P(m), o, a);
            } catch {
            }
          }
        }
      return "";
    }
    var n = Object.prototype.hasOwnProperty, i = {}, c = $.ReactDebugCurrentFrame;
    function l(r) {
      if (r) {
        var o = r._owner, a = e(r.type, r._source, o ? o.type : null);
        c.setExtraStackFrame(a);
      } else
        c.setExtraStackFrame(null);
    }
    function h(r, o, a, f, m) {
      {
        var P = Function.call.bind(n);
        for (var g in r)
          if (P(r, g)) {
            var p = void 0;
            try {
              if (typeof r[g] != "function") {
                var B = Error((f || "React class") + ": " + a + " type `" + g + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof r[g] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw B.name = "Invariant Violation", B;
              }
              p = r[g](o, g, f, a, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (S) {
              p = S;
            }
            p && !(p instanceof Error) && (l(m), A("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", f || "React class", a, g, typeof p), l(null)), p instanceof Error && !(p.message in i) && (i[p.message] = !0, l(m), A("Failed %s type: %s", a, p.message), l(null));
          }
      }
    }
    var C = Array.isArray;
    function x(r) {
      return C(r);
    }
    function T(r) {
      {
        var o = typeof Symbol == "function" && Symbol.toStringTag, a = o && r[Symbol.toStringTag] || r.constructor.name || "Object";
        return a;
      }
    }
    function H(r) {
      try {
        return vt(r), !1;
      } catch {
        return !0;
      }
    }
    function vt(r) {
      return "" + r;
    }
    function Bt(r) {
      if (H(r))
        return A("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", T(r)), vt(r);
    }
    var gt = $.ReactCurrentOwner, Kt = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Mt, Nt, kt;
    kt = {};
    function Zt(r) {
      if (n.call(r, "ref")) {
        var o = Object.getOwnPropertyDescriptor(r, "ref").get;
        if (o && o.isReactWarning)
          return !1;
      }
      return r.ref !== void 0;
    }
    function Qt(r) {
      if (n.call(r, "key")) {
        var o = Object.getOwnPropertyDescriptor(r, "key").get;
        if (o && o.isReactWarning)
          return !1;
      }
      return r.key !== void 0;
    }
    function te(r, o) {
      if (typeof r.ref == "string" && gt.current && o && gt.current.stateNode !== o) {
        var a = N(gt.current.type);
        kt[a] || (A('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', N(gt.current.type), r.ref), kt[a] = !0);
      }
    }
    function ee(r, o) {
      {
        var a = function() {
          Mt || (Mt = !0, A("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", o));
        };
        a.isReactWarning = !0, Object.defineProperty(r, "key", {
          get: a,
          configurable: !0
        });
      }
    }
    function re(r, o) {
      {
        var a = function() {
          Nt || (Nt = !0, A("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", o));
        };
        a.isReactWarning = !0, Object.defineProperty(r, "ref", {
          get: a,
          configurable: !0
        });
      }
    }
    var ne = function(r, o, a, f, m, P, g) {
      var p = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: _,
        // Built-in properties that belong on the element
        type: r,
        key: o,
        ref: a,
        props: g,
        // Record the component responsible for creating this element.
        _owner: P
      };
      return p._store = {}, Object.defineProperty(p._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(p, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: f
      }), Object.defineProperty(p, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: m
      }), Object.freeze && (Object.freeze(p.props), Object.freeze(p)), p;
    };
    function ie(r, o, a, f, m) {
      {
        var P, g = {}, p = null, B = null;
        a !== void 0 && (Bt(a), p = "" + a), Qt(o) && (Bt(o.key), p = "" + o.key), Zt(o) && (B = o.ref, te(o, m));
        for (P in o)
          n.call(o, P) && !Kt.hasOwnProperty(P) && (g[P] = o[P]);
        if (r && r.defaultProps) {
          var S = r.defaultProps;
          for (P in S)
            g[P] === void 0 && (g[P] = S[P]);
        }
        if (p || B) {
          var R = typeof r == "function" ? r.displayName || r.name || "Unknown" : r;
          p && ee(g, R), B && re(g, R);
        }
        return ne(r, p, B, m, f, gt.current, g);
      }
    }
    var At = $.ReactCurrentOwner, Vt = $.ReactDebugCurrentFrame;
    function ot(r) {
      if (r) {
        var o = r._owner, a = e(r.type, r._source, o ? o.type : null);
        Vt.setExtraStackFrame(a);
      } else
        Vt.setExtraStackFrame(null);
    }
    var Dt;
    Dt = !1;
    function It(r) {
      return typeof r == "object" && r !== null && r.$$typeof === _;
    }
    function Wt() {
      {
        if (At.current) {
          var r = N(At.current.type);
          if (r)
            return `

Check the render method of \`` + r + "`.";
        }
        return "";
      }
    }
    function oe(r) {
      {
        if (r !== void 0) {
          var o = r.fileName.replace(/^.*[\\\/]/, ""), a = r.lineNumber;
          return `

Check your code at ` + o + ":" + a + ".";
        }
        return "";
      }
    }
    var Yt = {};
    function ae(r) {
      {
        var o = Wt();
        if (!o) {
          var a = typeof r == "string" ? r : r.displayName || r.name;
          a && (o = `

Check the top-level render call using <` + a + ">.");
        }
        return o;
      }
    }
    function zt(r, o) {
      {
        if (!r._store || r._store.validated || r.key != null)
          return;
        r._store.validated = !0;
        var a = ae(o);
        if (Yt[a])
          return;
        Yt[a] = !0;
        var f = "";
        r && r._owner && r._owner !== At.current && (f = " It was passed a child from " + N(r._owner.type) + "."), ot(r), A('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', a, f), ot(null);
      }
    }
    function Gt(r, o) {
      {
        if (typeof r != "object")
          return;
        if (x(r))
          for (var a = 0; a < r.length; a++) {
            var f = r[a];
            It(f) && zt(f, o);
          }
        else if (It(r))
          r._store && (r._store.validated = !0);
        else if (r) {
          var m = tt(r);
          if (typeof m == "function" && m !== r.entries)
            for (var P = m.call(r), g; !(g = P.next()).done; )
              It(g.value) && zt(g.value, o);
        }
      }
    }
    function se(r) {
      {
        var o = r.type;
        if (o == null || typeof o == "string")
          return;
        var a;
        if (typeof o == "function")
          a = o.propTypes;
        else if (typeof o == "object" && (o.$$typeof === v || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        o.$$typeof === k))
          a = o.propTypes;
        else
          return;
        if (a) {
          var f = N(o);
          h(a, r.props, "prop", f, r);
        } else if (o.PropTypes !== void 0 && !Dt) {
          Dt = !0;
          var m = N(o);
          A("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", m || "Unknown");
        }
        typeof o.getDefaultProps == "function" && !o.getDefaultProps.isReactClassApproved && A("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ue(r) {
      {
        for (var o = Object.keys(r.props), a = 0; a < o.length; a++) {
          var f = o[a];
          if (f !== "children" && f !== "key") {
            ot(r), A("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", f), ot(null);
            break;
          }
        }
        r.ref !== null && (ot(r), A("Invalid attribute `ref` supplied to `React.Fragment`."), ot(null));
      }
    }
    function $t(r, o, a, f, m, P) {
      {
        var g = ct(r);
        if (!g) {
          var p = "";
          (r === void 0 || typeof r == "object" && r !== null && Object.keys(r).length === 0) && (p += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var B = oe(m);
          B ? p += B : p += Wt();
          var S;
          r === null ? S = "null" : x(r) ? S = "array" : r !== void 0 && r.$$typeof === _ ? (S = "<" + (N(r.type) || "Unknown") + " />", p = " Did you accidentally export a JSX literal instead of a component?") : S = typeof r, A("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", S, p);
        }
        var R = ie(r, o, a, m, P);
        if (R == null)
          return R;
        if (g) {
          var z = o.children;
          if (z !== void 0)
            if (f)
              if (x(z)) {
                for (var at = 0; at < z.length; at++)
                  Gt(z[at], r);
                Object.freeze && Object.freeze(z);
              } else
                A("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Gt(z, r);
        }
        return r === s ? ue(R) : se(R), R;
      }
    }
    function ce(r, o, a) {
      return $t(r, o, a, !0);
    }
    function le(r, o, a) {
      return $t(r, o, a, !1);
    }
    var fe = le, he = ce;
    mt.Fragment = s, mt.jsx = fe, mt.jsxs = he;
  }()), mt;
}
process.env.NODE_ENV === "production" ? jt.exports = de() : jt.exports = pe();
var Rt = jt.exports;
const Ft = {
  TOP_LEFT: 0,
  TOP_RIGHT: 1,
  BOTTOM_LEFT: 2,
  BOTTOM_RIGHT: 3
};
function ve() {
  var b = 100, _ = ["webkit", "moz", ""], D = "http://www.w3.org/2000/svg", s = "peel-", G, V, y, I, v, E = document.documentElement, L = E.style;
  function k(t) {
    for (var e, n, i = 0; i < _.length; i++)
      if (e = _[i], n = e ? e + A(t) : t, n in L)
        return n;
  }
  function W() {
    G = k("clipPath"), V = k("transform"), y = k("boxShadow"), I = k("filter"), q(), u.supported = !!(G && V), u.effectsSupported = v;
  }
  function q() {
    var t = document.createElement("div"), e = t.style;
    e.cssText = "background:linear-gradient(45deg,#9f9,white);", v = (e.backgroundImage || "").indexOf("gradient") > -1;
  }
  function w(t) {
    return Math.round(t * b) / b;
  }
  function M(t) {
    return Math.max(0, Math.min(1, t));
  }
  function tt(t, e, n) {
    return (t - e) / (n - e);
  }
  function $(t, e) {
    return (e || 1) * 2 * (0.5 - Math.abs(t - 0.5));
  }
  function A(t) {
    return t.slice(0, 1).toUpperCase() + t.slice(1);
  }
  function xt(t) {
    return t.replace(/-(\w)/g, function(e, n) {
      return n.toUpperCase();
    });
  }
  function X(t) {
    return s + t;
  }
  function Tt(t, e) {
    t.style[G] = e;
  }
  function st(t, e) {
    t.style[V] = e;
  }
  function _t(t, e, n, i, c, l) {
    t.style[y] = ut(e, n, i, c, l);
  }
  function Lt(t, e, n, i, c) {
    t.style[I] = "drop-shadow(" + ut(e, n, i, null, c) + ")";
  }
  function ut(t, e, n, i, c) {
    return w(t) + "px " + w(e) + "px " + w(n) + "px " + (i ? w(i) + "px " : "") + "rgba(0,0,0," + w(c) + ")";
  }
  function ct(t, e) {
    t.style.opacity = e;
  }
  function lt(t, e, n) {
    if (v) {
      var i;
      n.length === 0 ? i = "none" : i = "linear-gradient(" + w(e) + "deg," + n.join(",") + ")", t.style.backgroundImage = i;
    }
  }
  function nt(t, e, n, i) {
    t.addEventListener(e, n, i);
  }
  function N(t, e, n) {
    t.removeEventListener(e, n);
  }
  function U(t, e) {
    var n = t.changedTouches ? t.changedTouches[0] : t;
    return {
      x: n.clientX - e.getBoundingClientRect().left,
      y: n.clientY - e.getBoundingClientRect().top
    };
  }
  function j(t, e) {
    return ft(0, 0, 0, t, e);
  }
  function et(t, e) {
    return ft(255, 255, 255, t, e);
  }
  function ft(t, e, n, i, c) {
    return i = w(M(i)), "rgba(" + t + "," + e + "," + n + "," + i + ") " + w(c * 100) + "%";
  }
  function ht(t, e) {
    return typeof t == "string" && (t = (e || document).querySelector(t)), t;
  }
  function bt(t, e) {
    var n = document.createElement("div");
    return rt(n, e), t.appendChild(n), n;
  }
  function rt(t, e) {
    t.classList.add(e);
  }
  function Pt(t) {
    return t.style.zIndex;
  }
  function wt(t, e) {
    t.style.zIndex = e;
  }
  function J(t, e, n) {
    e = e || E;
    var i = document.createElementNS(D, t);
    e.appendChild(i);
    for (var c in n)
      Object.prototype.hasOwnProperty.call(n, c) && dt(i, c, n[c]);
    return i;
  }
  function dt(t, e, n) {
    t.setAttributeNS(null, e, n);
  }
  function u(t, e) {
    this.setOptions(e), this.el = ht(t, E), this.constraints = [], this.events = [], this.dragHandlers = [], this.setupLayers(), this.setupDimensions(), this.setCorner(this.getOption("corner")), this.setMode(this.getOption("mode")), this.init();
  }
  u.Corners = Ft, u.Defaults = {
    topShadow: !0,
    topShadowBlur: 5,
    topShadowAlpha: 0.5,
    topShadowOffsetX: 0,
    topShadowOffsetY: 1,
    topShadowCreatesShape: !0,
    backReflection: !1,
    backReflectionSize: 0.02,
    backReflectionOffset: 0,
    backReflectionAlpha: 0.15,
    backReflectionDistribute: !0,
    backShadow: !0,
    backShadowSize: 0.04,
    backShadowOffset: 0,
    backShadowAlpha: 0.1,
    backShadowDistribute: !0,
    bottomShadow: !0,
    bottomShadowSize: 1.5,
    bottomShadowOffset: 0,
    bottomShadowDarkAlpha: 0.7,
    bottomShadowLightAlpha: 0.1,
    bottomShadowDistribute: !0,
    setPeelOnInit: !0,
    clippingBoxScale: 4,
    flipConstraintOffset: 5,
    dragPreventsDefault: !0
  }, u.prototype.setCorner = function() {
    var t = arguments;
    t[0] === void 0 ? t = [u.Corners.BOTTOM_RIGHT] : t[0].length && (t = t[0]), this.corner = this.getPointOrCorner(t);
  }, u.prototype.setMode = function(t) {
    t === "book" ? (this.addPeelConstraint(u.Corners.BOTTOM_LEFT), this.addPeelConstraint(u.Corners.TOP_LEFT), this.setOption("backReflection", !1), this.setOption("backShadowDistribute", !1), this.setOption("bottomShadowDistribute", !1)) : t === "calendar" && (this.addPeelConstraint(u.Corners.TOP_RIGHT), this.addPeelConstraint(u.Corners.TOP_LEFT));
  }, u.prototype.setPeelPath = function(t, e) {
    var n = arguments, i, c, l, h;
    i = new d(t, e), n.length === 4 ? (c = new d(n[2], n[3]), this.path = new O(i, c)) : n.length === 8 && (l = new d(n[2], n[3]), h = new d(n[4], n[5]), c = new d(n[6], n[7]), this.path = new Et(i, l, h, c));
  }, u.prototype.handleDrag = function(t, e) {
    this.dragHandlers.push(t), this.setupDragEvents(e);
  }, u.prototype.handlePress = function(t, e) {
    this.pressHandler = t, this.setupDragEvents(e);
  }, u.prototype.setupDragEvents = function(t) {
    var e = this, n, i, c;
    if (this.dragEventsSetup)
      return;
    t = t || this.el;
    function l(T, H) {
      e.getOption("dragPreventsDefault") && H.preventDefault(), i = T ? "touchmove" : "mousemove", c = T ? "touchend" : "mouseup", nt(E, i, h), nt(E, c, C), n = !1;
    }
    function h(T) {
      e.dragHandlers.forEach((H) => {
        x(H, T);
      }), n = !0;
    }
    function C(T) {
      !n && e.pressHandler && x(e.pressHandler, T), N(E, i, h), N(E, c, C);
    }
    function x(T, H) {
      var vt = U(H, e.el);
      T.call(e, H, vt.x, vt.y);
    }
    this.addEvent(t, "mousedown", l.bind(this, !1)), this.addEvent(
      t,
      "touchstart",
      l.bind(this, !0),
      pt() ? {
        passive: !0
      } : !1
    ), this.dragEventsSetup = !0;
  };
  function pt() {
    let t = !1;
    try {
      const e = Object.defineProperty({}, "passive", {
        get: function() {
          return t = !0, null;
        }
      });
      window.addEventListener("test", null, e);
    } catch {
      console.warn("Passive event listener not support");
    }
    return t;
  }
  u.prototype.removeEvents = function() {
    this.events.forEach(function(t, e) {
      N(t.el, t.type, t.handler);
    }), this.events = [];
  }, u.prototype.removeGlobalSvgElements = function() {
    var t = document.getElementsByClassName(
      X("svg-clip-element")
    );
    [...t].forEach((e) => e.remove());
  }, u.prototype.clear = function() {
    this.removeEvents(), this.removeGlobalSvgElements();
  }, u.prototype.setTimeAlongPath = function(t) {
    t = M(t);
    var e = this.path.getPointForTime(t);
    this.timeAlongPath = t, this.setPeelPosition(e.x, e.y);
  }, u.prototype.setFadeThreshold = function(t) {
    this.fadeThreshold = t;
  }, u.prototype.setPeelPosition = function() {
    var t = this.getPointOrCorner(arguments);
    t = this.getConstrainedPeelPosition(t), t && (this.peelLineSegment = this.getPeelLineSegment(t), this.peelLineRotation = this.peelLineSegment.getAngle(), this.setClipping(), this.setBackTransform(t), this.setEffects());
  }, u.prototype.addPeelConstraint = function() {
    var t = this.getPointOrCorner(arguments), e = this.corner.subtract(t).getLength();
    this.constraints.push(new K(t, e)), this.calculateFlipConstraint();
  }, u.prototype.setOption = function(t, e) {
    this.options[t] = e;
  }, u.prototype.getOption = function(t) {
    return this.options[xt(t)];
  }, u.prototype.getAmountClipped = function() {
    var t = this.getTopClipArea(), e = this.width * this.height;
    return tt(t, e, 0);
  }, u.prototype.addEvent = function(t, e, n, i) {
    return nt(t, e, n, i), this.events.push({
      el: t,
      type: e,
      handler: n
    }), n;
  }, u.prototype.getTopClipArea = function() {
    var t = new Y();
    return this.elementBox.forEach(function(e) {
      this.distributeLineByPeelLine(e, t);
    }, this), Y.getArea(t.getPoints());
  }, u.prototype.calculateFlipConstraint = function() {
    var t = this.corner, e = this.constraints.concat();
    this.flipConstraint = e.sort(function(n, i) {
      return t.y - n.center.y, t.y - i.center.y, n - i;
    })[0];
  }, u.prototype.dragStart = function(t, e, n) {
  }, u.prototype.fireHandler = function(t, e) {
    var n = U(t, this.el);
    e.call(this, t, n.x, n.y);
  }, u.prototype.setClipping = function() {
    var t = new Y(), e = new Y();
    this.clippingBox.forEach(function(n) {
      this.distributeLineByPeelLine(n, t, e);
    }, this), this.topClip.setPoints(t.getPoints()), this.backClip.setPoints(e.getPoints());
  }, u.prototype.distributeLineByPeelLine = function(t, e, n) {
    var i = this.peelLineSegment.getIntersectPoint(t);
    this.distributePointByPeelLine(t.p1, e, n), this.distributePointByPeelLine(i, e, n);
  }, u.prototype.distributePointByPeelLine = function(t, e, n) {
    if (t) {
      var i = this.peelLineSegment.getPointDeterminant(t);
      i <= 0 && e.addPoint(t), i >= 0 && n && n.addPoint(this.flipPointHorizontally(t));
    }
  }, u.prototype.setOptions = function(t) {
    var e = t || {}, n = u.Defaults;
    for (var i in n)
      !Object.prototype.hasOwnProperty.call(n, i) || i in e || (e[i] = n[i]);
    this.options = e;
  }, u.prototype.findOrCreateLayer = function(t, e, n) {
    var i = t + "-element", c = X(t), l = ht(this.getOption(i) || "." + c, e);
    return l || (l = bt(e, c)), rt(l, X("layer")), wt(l, n), l;
  }, u.prototype.getPointOrCorner = function(t) {
    return t.length === 2 ? new d(t[0], t[1]) : typeof t[0] == "number" ? this.getCornerPoint(t[0]) : t[0];
  }, u.prototype.getCornerPoint = function(t) {
    var e = +!!(t & 1) * this.width, n = +!!(t & 2) * this.height;
    return new d(e, n);
  }, u.prototype.getOptionalShape = function() {
    var t = ["rect", "polygon", "path", "circle"], e;
    return t.some(function(n) {
      let i = this.getOption("shape");
      var l = i && i[n], h;
      return l && (h = {}, h.attributes = l, h.type = n, e = h), e;
    }, this), e;
  }, u.prototype.setupLayers = function() {
    var t = this.getOptionalShape(), e = this.topLayer = this.findOrCreateLayer(
      "top",
      this.el,
      2
    ), n = this.backLayer = this.findOrCreateLayer(
      "back",
      this.el,
      3
    );
    this.bottomLayer = this.findOrCreateLayer("bottom", this.el, 1), t ? (this.topLayer = this.wrapShapeLayer(this.topLayer, "top-outer-clip"), this.backLayer = this.wrapShapeLayer(this.backLayer, "back-outer-clip"), this.topShapeClip = new F(e, t), this.backShapeClip = new F(n, t), this.bottomShapeClip = new F(this.bottomLayer, t), this.getOption("topShadowCreatesShape") && (this.topShadowElement = this.setupDropShadow(t, e))) : this.topShadowElement = this.findOrCreateLayer(
      "top-shadow",
      e,
      1
    ), this.topClip = new F(this.topLayer), this.backClip = new F(this.backLayer), this.backShadowElement = this.findOrCreateLayer(
      "back-shadow",
      n,
      1
    ), this.backReflectionElement = this.findOrCreateLayer(
      "back-reflection",
      n,
      2
    ), this.bottomShadowElement = this.findOrCreateLayer(
      "bottom-shadow",
      this.bottomLayer,
      1
    ), this.usesBoxShadow = !t;
  }, u.prototype.setupDropShadow = function(t, e) {
    var n = J("svg", e, {
      class: X("layer")
    });
    return J(t.type, n, t.attributes), n;
  }, u.prototype.wrapShapeLayer = function(t, e) {
    var n = Pt(t);
    rt(t, X("shape-layer"));
    var i = this.findOrCreateLayer(e, this.el, n);
    return i.appendChild(t), i;
  }, u.prototype.setupDimensions = function() {
    this.width = this.el.offsetWidth, this.height = this.el.offsetHeight, this.center = new d(this.width / 2, this.height / 2), this.elementBox = this.getScaledBox(1), this.clippingBox = this.getScaledBox(this.getOption("clippingBoxScale"));
  }, u.prototype.getScaledBox = function(t) {
    var e = t, n = t - 1, i = new d(-this.width * n, -this.height * n), c = new d(this.width * e, -this.height * n), l = new d(this.width * e, this.height * e), h = new d(-this.width * n, this.height * e);
    return [
      new O(i, c),
      new O(c, l),
      new O(l, h),
      new O(h, i)
    ];
  }, u.prototype.getConstrainedPeelPosition = function(t) {
    return this.constraints.forEach(function(e) {
      var n = this.getFlipConstraintOffset(e, t);
      n && (e = new K(e.center, e.radius - n)), t = e.constrainPoint(t);
    }, this), t;
  }, u.prototype.getFlipConstraintOffset = function(t, e) {
    var n = this.getOption("flipConstraintOffset");
    if (t === this.flipConstraint && n) {
      var i = this.corner.subtract(this.center), c = this.corner.subtract(t.center), l = c.getAngle();
      c.rotate(-l);
      var h = e.subtract(t.center).rotate(-l);
      if (i.x * i.y < 0 && (h.y *= -1), h.x > 0 && h.y > 0)
        return tt(h.getAngle(), 45, 0) * n;
    }
  }, u.prototype.getPeelLineSegment = function(t) {
    var e = this.corner.subtract(t).scale(0.5), n = t.add(e);
    e.x === 0 && e.y === 0 && (e = t.subtract(this.center));
    var i = e.getLength(), c = Math.max(this.width, this.height) / i * 10, l = e.rotate(-90).scale(c), h = n.add(l), C = n.subtract(l);
    return new O(h, C);
  }, u.prototype.setBackTransform = function(t) {
    var e = this.flipPointHorizontally(this.corner), n = (this.peelLineRotation - 90) * 2, i = t.subtract(e.rotate(n)), c = "translate(" + w(i.x) + "px, " + w(i.y) + "px) rotate(" + w(n) + "deg)";
    st(this.backLayer, c), st(this.topShadowElement, c);
  }, u.prototype.getPeelLineDistance = function() {
    var t, e, n, i;
    this.peelLineRotation < 90 ? (t = u.Corners.TOP_RIGHT, e = u.Corners.BOTTOM_LEFT) : this.peelLineRotation < 180 ? (t = u.Corners.BOTTOM_RIGHT, e = u.Corners.TOP_LEFT) : this.peelLineRotation < 270 ? (t = u.Corners.BOTTOM_LEFT, e = u.Corners.TOP_RIGHT) : this.peelLineRotation < 360 && (t = u.Corners.TOP_LEFT, e = u.Corners.BOTTOM_RIGHT), n = this.getCornerPoint(t), i = this.getCornerPoint(e);
    var c = new O(n, i).scale(2), l = this.peelLineSegment.getIntersectPoint(c);
    if (!l)
      return 2;
    var h = n.subtract(l).getLength(), C = n.subtract(i).getLength();
    return h / C;
  }, u.prototype.setEffects = function() {
    var t = this.getPeelLineDistance();
    this.setTopShadow(t), this.setBackShadow(t), this.setBackReflection(t), this.setBottomShadow(t), this.setFade();
  }, u.prototype.setTopShadow = function(t) {
    if (this.getOption("topShadow")) {
      var e = this.getOption("topShadowBlur"), n = this.getOption("topShadowOffsetX"), i = this.getOption("topShadowOffsetY"), c = this.getOption("topShadowAlpha"), l = this.exponential(t, 5, c);
      this.usesBoxShadow ? _t(this.topShadowElement, n, i, e, 0, l) : Lt(this.topShadowElement, n, i, e, l);
    }
  }, u.prototype.distributeOrLinear = function(t, e, n) {
    return e ? $(t, n) : t * n;
  }, u.prototype.exponential = function(t, e, n) {
    return n * M(Math.pow(1 + t, e) - 1);
  }, u.prototype.setBackReflection = function(t) {
    var e = [];
    if (this.canSetLinearEffect("backReflection", t)) {
      var n = this.getOption("backReflectionDistribute"), i = this.getOption("backReflectionSize"), c = this.getOption("backReflectionOffset"), l = this.getOption("backReflectionAlpha"), h = this.distributeOrLinear(t, n, i), C = t - c, x = C - h, T = x - h;
      e.push(et(0, 0)), e.push(et(0, T)), e.push(et(l, x)), e.push(et(0, C));
    }
    lt(
      this.backReflectionElement,
      180 - this.peelLineRotation,
      e
    );
  }, u.prototype.setBackShadow = function(t) {
    var e = [];
    if (this.canSetLinearEffect("backShadow", t)) {
      var n = this.getOption("backShadowSize"), i = this.getOption("backShadowOffset"), c = this.getOption("backShadowAlpha"), l = this.getOption("backShadowDistribute"), h = this.distributeOrLinear(t, l, n), C = t - i, x = C - h, T = x - h;
      e.push(j(0, 0)), e.push(j(0, T)), e.push(j(c, x)), e.push(j(c, C));
    }
    lt(
      this.backShadowElement,
      180 - this.peelLineRotation,
      e
    );
  }, u.prototype.setBottomShadow = function(t) {
    var e = [];
    if (this.canSetLinearEffect("bottomShadow", t)) {
      var n = this.getOption("bottomShadowSize"), i = this.getOption("bottomShadowOffset"), c = this.getOption("bottomShadowDarkAlpha"), l = this.getOption("bottomShadowLightAlpha"), h = this.getOption("bottomShadowDistribute"), C = t - (0.025 - i), x = C - this.distributeOrLinear(t, h, 0.03) * n - i, T = x - (0.02 * n - i);
      e = [
        j(0, 0),
        j(0, T),
        j(l, x),
        j(l, C),
        j(c, t)
      ];
    }
    lt(
      this.bottomShadowElement,
      this.peelLineRotation + 180,
      e
    );
  }, u.prototype.canSetLinearEffect = function(t, e) {
    return this.getOption(t) && e > 0;
  }, u.prototype.setFade = function() {
    var t = this.fadeThreshold, e = 1, n;
    t && (this.timeAlongPath !== void 0 ? n = this.timeAlongPath : n = this.getAmountClipped(), n > t && (e = (1 - n) / (1 - t)), ct(this.topLayer, e), ct(this.backLayer, e), ct(this.bottomShadowElement, e));
  }, u.prototype.flipPointHorizontally = function(t) {
    return new d(t.x - (t.x - this.center.x) * 2, t.y);
  }, u.prototype.init = function() {
    this.getOption("setPeelOnInit") && this.setPeelPosition(this.corner), rt(this.el, X("ready"));
  };
  function F(t, e) {
    this.el = t, this.shape = F.createClipPath(
      t,
      e || {
        type: "polygon"
      }
    ), st(this.el, "translate(0px,0px)");
  }
  F.getDefs = function() {
    return this.defs || (this.svg = J("svg", null, {
      class: X("svg-clip-element")
    }), this.defs = J("defs", this.svg)), this.defs;
  }, F.createClipPath = function(t, e) {
    var n = F.getId(), i = J("clipPath", this.getDefs()), c = J(e.type, i, e.attributes);
    return dt(i, "id", n), Tt(t, "url(#" + n + ")"), c;
  };
  function it() {
    if (crypto && crypto.getRandomValues) {
      var t = new Uint32Array(4);
      return crypto.getRandomValues(t), t.join("").replace(/[^a-zA-Z0-9]/g, "");
    } else
      return Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
  }
  F.getId = function() {
    return "svg-clip-" + it();
  }, F.prototype.setPoints = function(t) {
    var e = t.map(function(n) {
      return w(n.x) + "," + w(n.y);
    }).join(" ");
    dt(this.shape, "points", e);
  };
  function K(t, e) {
    this.center = t, this.radius = e;
  }
  K.prototype.containsPoint = function(t) {
    if (this.boundingRectContainsPoint(t)) {
      var e = this.center.x - t.x, n = this.center.y - t.y;
      e *= e, n *= n;
      var i = e + n, c = this.radius * this.radius;
      return i <= c;
    }
    return !1;
  }, K.prototype.boundingRectContainsPoint = function(t) {
    return t.x >= this.center.x - this.radius && t.x <= this.center.x + this.radius && t.y >= this.center.y - this.radius && t.y <= this.center.y + this.radius;
  }, K.prototype.constrainPoint = function(t) {
    if (!this.containsPoint(t)) {
      var e = t.subtract(this.center).getAngle();
      t = this.center.add(new d(this.radius, 0).rotate(e));
    }
    return t;
  };
  function Y() {
    this.points = [];
  }
  Y.getArea = function(t) {
    var e = 0, n = 0;
    return t.forEach(function(i, c, l) {
      var h = l[(c + 1) % l.length];
      e += i.x * h.y, n += i.y * h.x;
    }), (e - n) / 2;
  }, Y.prototype.addPoint = function(t) {
    this.points.push(t);
  }, Y.prototype.getPoints = function() {
    return this.points;
  };
  function Et(t, e, n, i) {
    this.p1 = t, this.c1 = e, this.p2 = i, this.c2 = n;
  }
  Et.prototype.getPointForTime = function(t) {
    var e = Math.pow(1 - t, 3), n = 3 * t * Math.pow(1 - t, 2), i = 3 * Math.pow(t, 2) * (1 - t), c = Math.pow(t, 3), l = e * this.p1.x + n * this.c1.x + i * this.c2.x + c * this.p2.x, h = e * this.p1.y + n * this.c1.y + i * this.c2.y + c * this.p2.y;
    return new d(l, h);
  };
  function O(t, e) {
    this.p1 = t, this.p2 = e;
  }
  O.EPSILON = 1e-6, O.prototype.getPointForTime = function(t) {
    return this.p1.add(this.getVector().scale(t));
  }, O.prototype.scale = function(t) {
    var e = this.p1.add(this.p2.subtract(this.p1).scale(t)), n = this.p2.add(this.p1.subtract(this.p2).scale(t));
    return new O(e, n);
  }, O.prototype.getPointDeterminant = function(t) {
    var e = (t.x - this.p1.x) * (this.p2.y - this.p1.y) - (t.y - this.p1.y) * (this.p2.x - this.p1.x);
    return e > -O.EPSILON && e < O.EPSILON && (e = 0), e;
  }, O.prototype.getIntersectPoint = function(t) {
    var e = this;
    function n(T, H) {
      return T.x * H.y - T.y * H.x;
    }
    var i = e.p2.subtract(e.p1), c = t.p2.subtract(t.p1), l = n(t.p1.subtract(e.p1), i), h = n(i, c);
    if (h == 0)
      return null;
    var C = l / h, x = n(t.p1.subtract(e.p1), c) / h;
    return x >= 0 && x <= 1 && C >= 0 && C <= 1 ? e.p1.add(i.scale(x)) : null;
  }, O.prototype.getAngle = function() {
    return this.getVector().getAngle();
  }, O.prototype.getVector = function() {
    return this.vector || (this.vector = this.p2.subtract(this.p1)), this.vector;
  };
  function d(t, e) {
    this.x = t, this.y = e;
  }
  return d.DEGREES_IN_RADIANS = 180 / Math.PI, d.degToRad = function(t) {
    return t / d.DEGREES_IN_RADIANS;
  }, d.radToDeg = function(t) {
    for (var e = t * d.DEGREES_IN_RADIANS; e < 0; )
      e += 360;
    return e;
  }, d.vector = function(t, e) {
    var n = d.degToRad(t);
    return new d(Math.cos(n) * e, Math.sin(n) * e);
  }, d.prototype.add = function(t) {
    return new d(this.x + t.x, this.y + t.y);
  }, d.prototype.subtract = function(t) {
    return new d(this.x - t.x, this.y - t.y);
  }, d.prototype.scale = function(t) {
    return new d(this.x * t, this.y * t);
  }, d.prototype.getLength = function() {
    return Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2));
  }, d.prototype.getAngle = function() {
    return d.radToDeg(Math.atan2(this.y, this.x));
  }, d.prototype.setAngle = function(t) {
    return d.vector(t, this.getLength());
  }, d.prototype.rotate = function(t) {
    return this.setAngle(this.getAngle() + t);
  }, W(), u;
}
const ge = Ct(ye);
function ye({ children: b, options: _ = {}, containerProps: D, ...s }, G) {
  const V = Ut(null), y = Ut();
  be(), Q(() => {
    function v() {
      var k, W, q;
      if (!V.current)
        return;
      const E = ve(), L = new E(V.current, me(_));
      if (y.current = L, G && (G.current = L), s.peelPosition && y.current.setPeelPosition(
        s.peelPosition.x,
        s.peelPosition.y
      ), s.corner)
        if (Object.prototype.hasOwnProperty.call(s.corner, "x") && Object.prototype.hasOwnProperty.call(s.corner, "y")) {
          const { x: w, y: M } = s.corner;
          y.current.setCorner(w, M);
        } else
          St(s.corner) && y.current.setCorner(Ot(s.corner));
      s.constraints && (Array.isArray(s.constraints) ? s.constraints.forEach((w) => {
        I(w);
      }) : I(s.constraints)), s.mode && y.current.setMode(s.mode), s.fadeThreshold && y.current.setFadeThreshold(s.fadeThreshold), s.peelPath && Array.isArray(s.peelPath) && y.current.setPeelPath(...s.peelPath), s.drag && ((k = y.current) == null || k.handleDrag(function(w, M, tt) {
        y.current.setPeelPosition(M, tt);
      })), s.handleDrag && ((W = y.current) == null || W.handleDrag(
        (w, M, tt) => {
          var $;
          return ($ = s.handleDrag) == null ? void 0 : $.call(s, w, M, tt, y.current);
        }
      )), s.handlePress && ((q = y.current) == null || q.handlePress(
        (w) => {
          var M;
          return (M = s.handlePress) == null ? void 0 : M.call(s, w, y.current);
        }
      ));
    }
    return v(), () => {
      y.current && y.current.clear();
    };
  }, [_]), Q(() => {
    var v;
    s.timeAlongPath && ((v = y.current) == null || v.setTimeAlongPath(s.timeAlongPath));
  }, [s.timeAlongPath]), Q(() => {
    s.peelPosition && y.current.setPeelPosition(
      s.peelPosition.x,
      s.peelPosition.y
    );
  }, [s.peelPosition]), Q(() => {
    if (s.corner)
      if (Object.prototype.hasOwnProperty.call(s.corner, "x") && Object.prototype.hasOwnProperty.call(s.corner, "y")) {
        const { x: v, y: E } = s.corner;
        y.current.setCorner(v, E);
      } else
        St(s.corner) && y.current.setCorner(Ot(s.corner));
  }, [s.corner]), Q(() => {
    s.peelPath && Array.isArray(s.peelPath) && y.current.setPeelPath(...s.peelPath);
  }, [s.peelPath]), Q(() => {
    s.mode && y.current.setMode(s.mode);
  }, [s.mode]), Q(() => {
    s.fadeThreshold && y.current.setFadeThreshold(s.fadeThreshold);
  }, [s.fadeThreshold]), Q(() => {
    s.constraints && (Array.isArray(s.constraints) ? s.constraints.forEach((v) => {
      I(v);
    }) : I(s.constraints));
  }, [s.constraints]);
  function I(v) {
    if (St(v))
      y.current.addPeelConstraint(Ot(v));
    else if (Object.prototype.hasOwnProperty.call(v, "x") && Object.prototype.hasOwnProperty.call(v, "y")) {
      const E = v;
      y.current.addPeelConstraint(E.x, E.y);
    }
  }
  return /* @__PURE__ */ Rt.jsx(
    "div",
    {
      ref: V,
      className: "peel " + (s == null ? void 0 : s.className),
      ...D,
      style: {
        height: s.height || "100%",
        width: s.width || "100%",
        ...D == null ? void 0 : D.style
      },
      children: b
    }
  );
}
function me(b) {
  return b.corner && (b.corner = St(b.corner) ? Ot(b.corner) : void 0), b;
}
function St(b) {
  return Object.keys(Ft).includes(b);
}
function Ot(b) {
  return Ft[b];
}
ge.displayName = "PeelWrapper";
const be = () => {
  Q(() => {
    const b = document.createElement("style");
    return b.innerHTML = `
    .peel {
      position: relative;
      opacity: 0;
    }
    
    .peel-ready {
      opacity: 1;
    }
    
    .peel-svg-clip-element {
      position: absolute;
      top: -10000px;
      left: -10000px;
      width: 1px;
      height: 1px;
      opacity: 0;
    }
    
    .peel-layer {
      position: absolute;
      z-index: 1;
      width: 100%;
      height: 100%;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      -webkit-user-select: none;
      -moz-user-select: none;
      user-select: none;
      -webkit-transform-origin: top left;
      -moz-transform-origin: top left;
      transform-origin: top left;
    }
    `, document.head.appendChild(b), () => {
      document.head.removeChild(b);
    };
  }, []);
}, we = Ct(function({ className: b, ..._ }, D) {
  return /* @__PURE__ */ Rt.jsx("div", { ref: D, className: "peel-top " + b, ..._ });
}), Ee = Ct(function({ className: b, ..._ }, D) {
  return /* @__PURE__ */ Rt.jsx("div", { ref: D, className: "peel-back " + b, ..._ });
}), Se = Ct(function({ className: b, ..._ }, D) {
  return /* @__PURE__ */ Rt.jsx("div", { ref: D, className: "peel-bottom " + b, ..._ });
});
export {
  Ee as PeelBack,
  Se as PeelBottom,
  we as PeelTop,
  ge as PeelWrapper
};
